dias/analyzers/source_spectra_analyzer.py

```diff
@@ -0,0 +1,784 @@
+
+            # Refresh the database
+            self.refresh_data_index()
+
+            cursor = self.data_index.cursor()
+            query = "SELECT stop FROM files ORDER BY stop DESC LIMIT 1"
+            results = list(cursor.execute(query))
+            if results:
+                query_start_time = results[0][0]
+            else:
+                query_start_time = query_stop_time - self.period
+
+            self.logger.info(
+                "Searching for transits from %s to %s"
+                % (str(query_start_time), str(query_stop_time))
+            )
+
+            # nsigma distance in degree from transit from peak
+            time_delta = (
+                2
+                * self.nsigma
+                * cal_utils.guess_fwhm(
+                    400.0, pol="X", dec=src_body.dec.radians, sigma=True, seconds=True
+                )
+            )
+
+            # Find all calibration files that have transit of given source
+            f = self.Finder()
+            f.set_time_range(query_start_time, query_stop_time)
+            f.accept_all_global_flags()
+            f.only_corr()
+            f.filter_acqs((data_index.ArchiveInst.name == self.acq_suffix))
+            f.include_transits(src_body, time_delta=time_delta)
+            file_list = f.get_results()
+            nfile = len(file_list)
+            times = [file_list[ii][1] for ii in range(0, nfile)]
+
+            try:
+                all_files = file_list[0][0]
+                if not all_files:
+                    raise IndexError()
+            except IndexError:
+                tmp = "No {} files found from last {} for \
+                source transit {}.\n".format(
+                    self.acq_suffix, self.period, src
+                )
+                err_msg += tmp
+                self.logger.info(tmp)
+                continue
+
+            # Loop over files
+            for file_index, files in enumerate(all_files):
+                # Read file time range
+                with h5py.File(files, "r") as handler:
+                    file_timestamp = handler["index_map"]["time"]["ctime"][:]
+
+                csd = int(np.median(ephemeris.csd(file_timestamp)))
+
+                # Compute source coordinates
+                src_ra, src_dec = ephemeris.object_coords(
+                    src_body, date=np.median(file_timestamp), deg=False
+                )
+                source_start_time, source_stop_time = file_list[file_index][1]
+
+                is_daytime = 0.0
+                # test if the source is transiting in the daytime
+                solar_rise = ephemeris.solar_rising(
+                    source_start_time - 24.0 * 3600.0, end_time=source_stop_time
+                )
+                for rr in solar_rise:
+                    ss = ephemeris.solar_setting(rr)[0]
+                    if (source_start_time <= ss) and (rr <= source_stop_time):
+                        is_daytime += 1
```

Comment:
If you use strings for this config variable (let's call it `skip_transits`), you could do>
```
def is_skipped(start, stop):
  if self.skip_transits == "daytime" or self.skip_transits == "solar_transit":
    if (source_start_time <= ss) and (rr <= source_stop_time):
      return True
  if self.skip_transits == "solar_transit:
    if source_start_time <= tt <= source_stop_time:
      return True:
  return False

if is_skipped(start, stop, ss, rr):
  self.logger.info(...)
  continue
```

But maybe I am getting something wrong here. Should there ever be more than one entry in `solar_rise`? I understand it's the sunrises during the last 24h before the transit. If not, it could simplify this code to sanity check once that there is exactly one sunset/rise and then check similar to how I wrote above instead of looping over the one sunrise.

Reply:
I think there is a scenario where the sun rises between `source_start_time` and `source_stop_time`.  In that case,  `solar_rise` would be a two element array where the first element will evaluate false for both checks but the second element will evaluate True for the first check.

Maybe this could be resolved by using

```solar_rise = ephemeris.solar_rising(source_stop_time - 24.0 * 3600.0, end_time=source_stop_time)```

I think that results in a single element in `solar_rise` that covers all scenarios, but could be mistaken.